### ISSUE TITLE
#Crash: Android 4.4.4 setTranslationZ => java.lang.NoSuchMethodError

### ISSUE THREAD

**Issue by FredeHo** [opentok/cordova-plugin-opentok#77]
Hi guys,

I'm experiencing a crash in my Ionic-App on an Android 4.4.4 Tablet.
` java.lang.NoSuchMethodError: android.view.View.setTranslationZ`

This isn't a big suprise since there seem to be a few _setTranslationZ_ calls, which are only supported since API level 21.

Is there any fallback for devices < Android 21? Or would a 
`if (android.os.Build.VERSION.SDK_INT >= 21)`
around the setTranslationZ calls do the trick? I assume this would probably break the Z positioning(?)

EDIT:
We're using crosswalk and the cordova-plugin-opentok version 3.2.1

Best regards

**Comment by mark-veenstra** [comment]
We build a seperate APK for API levels below 21. We use crosswalk plugin
for those older builds.

Mark

Op do 19 apr. 2018 om 15:19 schreef Frederik Hobein <
notifications@github.com>:

**Comment by FredeHo** [comment]
Yeah we build seperate APKs as well. I should have mentioned it. We're using crosswalk and the newest cordova-plugin-opentok version (3.2.1)

**Comment by FredeHo** [comment]
I can confirm now: Wrapping all setTranslationZ calls in OpenTokAndroidPlugin.java around like this e.g.
```
if(android.os.Build.VERSION.SDK_INT >= 21) {
     viewContainer.mView.setTranslationZ(viewContainer.getZIndex());
}
```
did fix the crashes. Working now. Might be a bit hacky, bit for now its fine!

**Comment by wolfenrain** [comment]
@FredeHo Are you doing negative valued `z-index`? I am currently working on it(may need some changes) and was wondering if you are having any issues with negative values

**Comment by msach22** [comment]
Released with `3.3.0`
